sdks/typescript/packages/sdk/src/table_cache.ts

```diff
@@ -57,10 +65,21 @@ export class TableCache<RowType = any> {
   /**
    * @returns The values of the rows in the table
    */
-  iter(): any[] {
+  iter(): RowType[] {
     return Array.from(this.rows.values()).map(([row]) => row);
   }
 
+  remoteQuery(ctx: DbContext, filters: string): Promise<RowType[]> {
+    return new Promise((resolve, reject) => {
+      const name = this.name();
+
+      ctx.queryBuilder()
```

Comment:
As you might've noticed, this requires having `ctx` passed in as an argument, as we've nowhere to find it from here. Suggestions are welcomed.

Reply:
You will need to modify the table handle objects so that they contain references to everything in `ctx` necessary to complete this operation. This is likely to be easier if you bypass your `queryBuilder` interface, which will not be part of the public API anyways.

Reply:
I've implemented this the least invasive way, not requiring additional changes to the codegen. It is possible, though, to pass `ctx` at a later stage (not storing the reference to it in `ClientCache`), with a subtle change in the codegen (namely, in `RemoteTables` gen), if you wish so better.